### PR TITLE
added cocoapods support

### PR DESCRIPTION
**Pull request by cvasilak**
done for [AGIOS-309](https://issues.jboss.org/browse/AGIOS-309)

demos have been updated to use the Podspec

Helloworld ->  https://github.com/aerogear/aerogear-push-helloworld/pull/77
Contacts    -> https://github.com/aerogear/aerogear-push-quickstarts/pull/122

### CODE REVIEW

**Comment by corinnekrych** [comment]
Installed and run ok too
the pb is the consistency, as we said AeroGear default impl is Swift, appending Swift to the podspec is confusing. maybe we should do the other way around append objc.
wdyt @cvasilak ?

File: AeroGearPushSwift.podspec [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  s.name         = "AeroGearPushSwift"
+  s.version      = "1.0.0"
+  s.summary      = "AeroGear UnifiedPush Client Registration SDK (Swift)."
+  s.homepage     = "https://github.com/aerogear/aerogear-ios-push"
+  s.license      = 'Apache License, Version 2.0'
+  s.author       = "Red Hat, Inc."
+  s.source       = { :git => 'https://github.com/aerogear/aerogear-ios-push.git', :branch => 'master' }
+  s.platform     = :ios, 8.0
+  s.source_files = 'push-sdk-swift/*.{swift}'  
+end
```
Comment (cvasilak) [inline AeroGearPushSwift.podspec]: it's implicit but never hurts to add it so I did

**Comment by cvasilak** [comment]
@corinnekrych @matzew have updated the PR based on my research on the naming issue and have written down the details on an issue opened on the [cocoapods site](https://github.com/CocoaPods/CocoaPods/issues/3025). Would like to see if that is the recommended approach or something else which I possible miss. I will wait a bit for their response

**Comment by cvasilak** [comment]
@danbev we follow a consistent readme instructions for adding the library either through cocoapods or with the git submodule (plus drag-and-drop). It's a theme followed by other swift libs mostly

**Comment by danbev** [comment]
@cvasilak Ah gotcha! I'll do the same for the sync libraries as well. Thx

**Comment by cvasilak** [comment]
as discussed on the ML[1] will go with the approach outlined in this PR

http://aerogear-dev.1069024.n5.nabble.com/aerogear-dev-Parallel-support-for-a-Cocoapods-Swift-Obj-c-library-tc10632.html

**Comment by cvasilak** [comment]
thank you @matzew @corinnekrych :rocket: landed on https://github.com/aerogear/aerogear-ios-push/commit/15ce66b1e7851c8e35788b97f70b7322d6798149

**Pull request merged**